amd_openvx_extensions/amd_rpp/source/CropMirrorNormalize.cpp

```diff
@@ -21,127 +21,111 @@ THE SOFTWARE.
-    rppHandle_t rppHandle;
-    Rpp32u device_type;
-    Rpp32u nbatchSize;
+struct CropMirrorNormalizeLocalData {
+    RPPCommonHandle * handle;
+    Rpp32u deviceType;
     RppPtr_t pSrc;
     RppPtr_t pDst;
-    vx_uint32 *start_x;
-    vx_uint32 *start_y;
-    vx_uint32 *crop_h;
-    vx_uint32 *crop_w;
-    vx_float32 *mean;
-    vx_float32 *std_dev;
+    vx_float32 *multiplier;
+    vx_float32 *offset;
     vx_uint32 *mirror;
-    RpptDescPtr src_desc_ptr;
+    RpptDescPtr srcDescPtr;
     RpptDesc srcDesc;
     RpptDesc dstDesc;
-    RpptDescPtr dst_desc_ptr;
-    RpptROI *roi_tensor_ptr;
+    RpptDescPtr dstDescPtr;
+    void *roiTensorPtr;
+    RpptROI *roiPtr;
     RpptRoiType roiType;
-    Rpp32u layout;
-    size_t in_tensor_dims[NUM_OF_DIMS];
-    size_t out_tensor_dims[NUM_OF_DIMS];
-    vx_enum in_tensor_type;
-    vx_enum out_tensor_type;
+    Rpp32s inputLayout;
+    Rpp32s outputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t ouputTensorDims[RPP_MAX_TENSOR_DIMS];
+    vx_enum inputTensorType;
+    vx_enum outputTensorType;
 #if ENABLE_HIP
     void *pSrc_dev;
     void *pDst_dev;
-    RpptROI *roi_tensor_ptr_dev;
 #endif
 };
 
-static vx_status VX_CALLBACK refreshCropMirrorNormalize(vx_node node, const vx_reference *parameters, vx_uint32 num, CropMirrorNormalizeLocalData *data)
-{
+static vx_status VX_CALLBACK refreshCropMirrorNormalize(vx_node node, const vx_reference *parameters, vx_uint32 num, CropMirrorNormalizeLocalData *data) {
     vx_status status = VX_SUCCESS;
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[1], 0, data->nbatchSize * 4, sizeof(unsigned), data->roi_tensor_ptr, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[4], 0, data->nbatchSize, sizeof(vx_uint32), data->crop_w, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[5], 0, data->nbatchSize, sizeof(vx_uint32), data->crop_h, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[6], 0, data->nbatchSize, sizeof(vx_uint32), data->start_x, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[7], 0, data->nbatchSize, sizeof(vx_uint32), data->start_y, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[8], 0, data->nbatchSize * 3, sizeof(vx_float32), data->mean, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[9], 0, data->nbatchSize * 3, sizeof(vx_float32), data->std_dev, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[10], 0, data->nbatchSize, sizeof(vx_uint32), data->mirror, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[3], 0, data->srcDescPtr->n * data->srcDescPtr->c, sizeof(vx_float32), data->multiplier, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
```

Comment:
are we going to use pinned memory for these parameters as we discussed before?

Reply:
Will work on changes required for ParameterVX and use pinned memory for the parameters. Will issue this as a separate PR in future